smoke-test/src/test/java/org/opennms/smoketest/sentinel/ClassificationEngineReloadIT.java

```diff
@@ -0,0 +1,193 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019-2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.smoketest.sentinel;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.preemptive;
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.opennms.netmgt.flows.rest.classification.GroupDTO;
+import org.opennms.netmgt.flows.rest.classification.RuleDTO;
+import org.opennms.netmgt.flows.rest.classification.RuleDTOBuilder;
+import org.opennms.plugins.elasticsearch.rest.SearchResultUtils;
+import org.opennms.smoketest.containers.OpenNMSContainer;
+import org.opennms.smoketest.stacks.IpcStrategy;
+import org.opennms.smoketest.stacks.NetworkProtocol;
+import org.opennms.smoketest.stacks.OpenNMSStack;
+import org.opennms.smoketest.stacks.StackModel;
+import org.opennms.smoketest.telemetry.Packets;
+import org.opennms.smoketest.utils.KarafShell;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestClientFactory;
+import io.searchbox.client.config.HttpClientConfig;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchResult;
+
+// Verifies that the classification engine is reloaded periodically on sentinel.
+// See NMS-12259 for more details
+public class ClassificationEngineReloadIT {
+
+    @Rule
+    public Timeout timeout = new Timeout(20, TimeUnit.MINUTES);
+
+    @Rule
+    public final OpenNMSStack stack = OpenNMSStack.withModel(StackModel.newBuilder()
+            .withMinion()
+            .withSentinel()
+            .withIpcStrategy(IpcStrategy.JMS)
+            .withTelemetryProcessing()
+            .build());
+
+    @Test
```

Comment:
Can we move this to an existing class instead of creating a new one?

Reply:
Yeah the idea was to put it somewhere else, but requires changes to the existing tests, so I decided to put it here for now, until the test passes.